File: src/com/mapdigit/game/box2d/actor/RainDropGroup.scala
package com.mapdigit.game.box2d.actor

import com.guidebee.game.GameEngine._
import com.guidebee.game.audio.Sound
import com.guidebee.game.physics.{Shape, BodyDef}
import com.guidebee.math.geometry.Rectangle
import com.guidebee.math.MathUtils
import com.guidebee.game.scene.{Actor, Group}
import com.guidebee.utils.TimeUtils


class RainDropGroup extends Group("RainDropGroup") {
  private var lastDropTime: Long = 0
  private val dropSound = assetManager.get("drop.wav", classOf[Sound])



  spawnRaindrop

  private def spawnRaindrop {
    val raindrop = new RainDrop()
    raindrop.setPosition(MathUtils.random(0, 800 - 64),480)
    val dropRect=new Rectangle(12,0,40,40)
    raindrop.initBody(BodyDef.BodyType.DynamicBody,Shape.Type.Circle, dropRect,1.0f,0,0f)
    addActor(raindrop)
    lastDropTime = TimeUtils.nanoTime()
  }


  def collisionDetected(actor:Actor): Unit ={
    dropSound.play()
    removeActor(actor)

  }

  override def act(delta: Float): Unit = {
    // check if we need to create a new raindrop
    if (TimeUtils.nanoTime() - lastDropTime > 1000000000) spawnRaindrop


    val rainDrops=getChildren().toArray[RainDrop](classOf[RainDrop])
    for(raindrop <- rainDrops){
     // val y=raindrop.getY() -200 * graphics.getDeltaTime()
     // raindrop.setY(y)
      val y=raindrop.getY()
      if (y + 64 < 0) {
        removeActor(raindrop)

      }

    }
  }

}

File: src/com/mapdigit/game/tutorial/drop/actor/RainDropGroup.java
package com.mapdigit.game.tutorial.drop.actor;

import com.guidebee.game.physics.BodyDef;
import com.guidebee.game.physics.Shape;
import com.guidebee.game.scene.Group;
import com.guidebee.math.MathUtils;
import com.guidebee.math.geometry.Rectangle;
import com.guidebee.utils.TimeUtils;

public class RainDropGroup extends Group {

    private long lastDropTime =0;



    private void spawnRainDrop(){
        RainDrop rainDrop=new RainDrop();
        //rainDrop.rotateBy((float)(Math.random()*360));
        //rainDrop.debug();

        rainDrop.setPosition(MathUtils.random(0,800-64),480);
        Rectangle dropRect=new Rectangle(0,0,32,32);
        rainDrop.initBody(BodyDef.BodyType.DynamicBody, Shape.Type.Circle, dropRect,1.0f,0,0.1f);
        addActor(rainDrop);
        lastDropTime= TimeUtils.nanoTime();
    }




    @Override
    public void act(float delta){
        if(TimeUtils.nanoTime()-lastDropTime>1000000000) {
            spawnRainDrop();
        }

        RainDrop [] rainDrops=getChildren().toArray(RainDrop.class);
        for(RainDrop rainDrop: rainDrops){
            float y = rainDrop.getY();

            if(y+64 <0){

                removeActor(rainDrop);
            }

        }
    }


}
